File: app/frontend/src/types/ChatEvent.ts
import type { Proposal } from "./Proposal";

export type ChatEventType = "text" | "tool" | "proposal" | "error" | "done";

/** One SSE frame. Only the fields the frame's type uses are populated. */
export interface ChatEvent {
  type: ChatEventType;
  text?: string | null;
  name?: string | null;
  detail?: string | null;
  state?: string | null;
  proposal?: Proposal | null;
  code?: string | null;
  message?: string | null;
  turns?: number | null;
  total_cost_usd?: number | null;
}

File: app/backend/src/tv_watchlist/models/chat_event.py
"""One frame of the chat stream, serialised once per SSE data line."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

from tv_watchlist.models.proposal import Proposal

ChatEventType = Literal["text", "tool", "proposal", "error", "done"]


class ChatEvent(BaseModel):
    """A single streamed frame; only the fields its type uses are populated."""

    type: ChatEventType
    text: str | None = None
    name: str | None = None
    detail: str | None = None
    state: str | None = None
    proposal: Proposal | None = None
    code: str | None = None
    message: str | None = None
    turns: int | None = None
    total_cost_usd: float | None = None
